examples/notebooks/blog/unittests_in_beam.ipynb

```diff
@@ -0,0 +1,359 @@
+      ],
+      "metadata": {
+        "id": "Z8__izORM3r8"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Fake client to simulate an external call\n",
+        "\n",
+        "import time\n",
+        "class Client():\n",
+        "   def get_data(self, api):\n",
+        "      time.sleep(3)\n",
+        "      return [0,1,2,3,4,5,6,7,8,9]\n",
+        "\n",
+        "MyApiCall = Client()"
+      ],
+      "metadata": {
+        "id": "GGPF7cY3Ntyj"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "#The following packages are used to run the example pipelines\n",
+        "\n",
+        "import apache_beam as beam\n",
+        "from apache_beam.io import ReadFromText, WriteToText\n",
+        "from apache_beam.options.pipeline_options import PipelineOptions\n",
+        "\n",
+        "class MyDoFn(beam.DoFn):\n",
+        "  def process(self,element):\n",
+        "          returned_record = MyApiCall.get_data(\"http://my-api-call.com\")\n",
+        "          if len(returned_record)!=10:\n",
+        "            raise ValueError(\"Length of record does not match expected length\")\n",
+        "          yield returned_record\n",
+        "\n",
+        "with beam.Pipeline() as p:\n",
+        "  result = (\n",
+        "          p\n",
+        "          | ReadFromText(\"/content/sample_data/anscombe.json\")\n",
+        "          | beam.ParDo(MyDoFn())\n",
+        "          | WriteToText(\"/content/example1\")\n",
+        "  )"
+      ],
+      "metadata": {
+        "id": "Ktk9EVIFzGfP"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "**Mocking Example**\n",
+        "\n",
+        "The following blocks of code illustrate how we can mock an API response, to test out the error message we've written. Note that we can use mocking to avoid making the actual API call in our test."
+      ],
+      "metadata": {
+        "id": "58GVMyMa2PwE"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "!pip install mock  # Install the 'mock' module"
+      ],
+      "metadata": {
+        "id": "ESclJ_G-6JcW"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# We import the mock package for mocking functionality.\n",
+        "from unittest.mock import Mock,patch\n",
+        "# from MyApiCall import get_data\n",
+        "import mock\n",
+        "\n",
+        "\n",
+        "# MyApiCall is a function that calls get_data to fetch some data via an API call.\n",
+        "@patch('MyApiCall.get_data')\n",
+        "def test_error_message_wrong_length(self, mock_get_data):\n",
```

Comment:
I noticed we're not actually invoking these tests ever, so this isn't really getting shown off. I think it is probably fine, but maybe worth calling out in a comment (something like `# tests not invoked by colab, but could be run using https://docs.pytest.org/en/stable/`